src/systems/user_commands/UserCommands.cc

```diff
@@ -953,6 +966,56 @@ bool UserCommandsPrivate::VisualService(const msgs::Visual &_req,
   return true;
 }
 
+//////////////////////////////////////////////////
+void UserCommandsPrivate::OnCmdMaterialColor(const msgs::MaterialColor &_msg)
+{
+  msgs::Visual _req;
+  int numberOfEntities = 0;
+  auto entities = entitiesFromScopedName(_msg.entity().name(),
+      *this->iface->ecm);
```

Comment:
Accessing the ECM here is not thread-safe because `OnCmdMaterialColor` is called from the transport thread. one thought is to create a new constructor for `VisualCommand` that takes a `MaterialColor` and check the type of message in `VisualCommand::Execute`: https://github.com/gazebosim/gz-sim/blob/cc4914adefdcb40d63178587cdf17f648ae38ac6/src/systems/user_commands/UserCommands.cc#L1726